crates/autopilot/src/infra/persistence/auction/mod.rs

```diff
@@ -0,0 +1,44 @@
+use {crate::domain, tracing::Instrument};
+
+pub mod dto;
+
+impl super::Persistence {
+    /// Persists the given auction in a background task.
+    pub fn store_auction(&self, id: domain::AuctionId, instance: &domain::Auction) {
+        if let Some(uploader) = self.s3.clone() {
+            let instance = dto::from_domain(instance.clone());
+            tokio::spawn(
+                async move {
+                    match uploader.upload(id.to_string(), &instance).await {
+                        Ok(key) => {
+                            tracing::info!(?key, "uploaded auction to s3");
+                        }
+                        Err(err) => {
+                            tracing::warn!(?err, "failed to upload auction to s3");
+                        }
+                    }
+                }
+                .instrument(tracing::Span::current()),
+            );
+        }
+    }
+
+    /// There is always only one `current` auction.
+    ///
+    /// This method replaces the current auction with the given one.
+    pub async fn replace_current_auction(
+        &self,
+        auction: domain::Auction,
+    ) -> Result<domain::AuctionId, Error> {
+        self.postgres
+            .replace_current_auction(&dto::from_domain(auction))
+            .await
+            .map_err(Error::DbError)
+    }
```

Comment:
I am not into the DDD yet, but it's not obvious that the two functions serve completely different purposes. Maybe better function naming would help.

Reply:
True, using same interface to store to both s3 and database can be ambiguous. Maybe renaming the `store_auction` to `archive_auction` is better?

Reply:
Yes, or to keep `replace_current_auction` as it is since it's not just about storing.

Reply:
Resolved.